cmd/utils/git.go

```diff
@@ -40,3 +40,38 @@ func GetBranch(ctx context.Context, path string) (string, error) {
 	name := strings.TrimPrefix(branch.String(), "refs/heads/")
 	return name, nil
 }
+
+func GetGitCommit(ctx context.Context, path string) (string, error) {
+	repo, err := git.PlainOpen(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to analyze git repo: %w", err)
+	}
+
+	head, err := repo.Head()
+	if err != nil {
+		return "", fmt.Errorf("failed to infer the git repo's current branch: %w", err)
+	}
+
+	hash := head.Hash()
+
+	return hash.String(), nil
+}
+
+func IsCleanDirectory(ctx context.Context, path string) (bool, error) {
+	repo, err := git.PlainOpen(path)
+	if err != nil {
+		return false, fmt.Errorf("failed to analyze git repo: %w", err)
+	}
+
+	worktree, err := repo.Worktree()
+	if err != nil {
+		return false, fmt.Errorf("failed to infer the git repo's current branch: %w", err)
+	}
+
+	status, err := worktree.Status()
+	if err != nil {
+		return false, fmt.Errorf("failed to infer the git repo's status: %w", err)
+	}
+
+	return status.IsClean(), nil
```

Comment:
is it possible to get a hash of the changes here?

Reply:
no without committing the changes, because timestamp is part of that hash. I have generated a random string